x-pack/test/functional/services/ml/job_wizard_common.ts

```diff
@@ -335,6 +335,37 @@ export function MachineLearningJobWizardCommonProvider(
       }
     },
 
+    async assertStartDatafeedSwitchExists() {
+      const subj = 'mlJobWizardStartDatafeedCheckbox';
+      await testSubjects.existOrFail(subj, { allowHidden: true });
+    },
+
+    async getStartDatafeedSwitchCheckedState(): Promise<boolean> {
+      const subj = 'mlJobWizardStartDatafeedCheckbox';
+      const isSelected = await testSubjects.getAttribute(subj, 'aria-checked');
+      return isSelected === 'true';
+    },
+
+    async assertStartDatafeedSwitchCheckedState(expectedValue: boolean) {
+      const actualCheckedState = await this.getStartDatafeedSwitchCheckedState();
+      expect(actualCheckedState).to.eql(
+        expectedValue,
+        `Expected start datafeed switch to be '${expectedValue ? 'enabled' : 'disabled'}' (got '${
+          actualCheckedState ? 'enabled' : 'disabled'
+        }')`
+      );
+    },
+
+    async toggleStartDatafeedSwitch(toggle: boolean) {
+      const subj = 'mlJobWizardStartDatafeedCheckbox';
+      if ((await this.getStartDatafeedSwitchCheckedState()) === !toggle) {
```

Comment:
nit:
```suggestion
      if ((await this.getStartDatafeedSwitchCheckedState()) !== toggle) {
```

Reply:
Updated in https://github.com/elastic/kibana/pull/77484/commits/fb025a122fd3adf4868883847143f05c961db5cf